# U.S. Medical Insurance Costs

Import the dataset. 

In [2]:
import csv

with open('/Users/ericarikan/Coding_Projects/python-portfolio-project-starter-files/insurance.csv', newline='') as insurance_file:
    insurance_reader = csv.DictReader(insurance_file)
    

Create lists for each attribute in the dataset, and populate them with the data from the file. 

In [3]:

age = []
sex = []
bmi = []
children = []
smoker = []
region = []
charges = []

Populate the lists. 

In [4]:
with open('/Users/ericarikan/Coding_Projects/python-portfolio-project-starter-files/insurance.csv', newline='') as insurance_file:
    insurance_reader = csv.DictReader(insurance_file, delimiter=',')
    for row in insurance_reader:
        age.append(row['age'])
        sex.append(row['sex'])
        bmi.append(row['bmi'])
        children.append(row['children'])
        smoker.append(row['smoker'])
        region.append(row['region'])
        charges.append(row['charges'])

The lists for `age`, `bmi`, `children`, and `charges` contain string values. They need to be converted to int or float types. Create functions to convert the list values, and use the functions to update the lists. 

In [5]:
def convert_to_int(list):
    updated_list = []
    for entry in list:
        new_entry = int(entry)
        updated_list.append(new_entry)
    return updated_list

def convert_to_float(list):
    updated_list = []
    for entry in list:
        new_entry = float(entry)
        updated_list.append(new_entry)
    return updated_list

# Convert age list to int
age_int = convert_to_int(age)

# Convert bmi list to float
bmi_float = convert_to_float(bmi)

# Convert children to int
children_int = convert_to_int(children)

# Convert charges to float
charges_float = convert_to_float(charges)

The lists are now populated. Create a dictionary that includes all values, with a unique ID for every entry. 

In [28]:
def populate_dict(list1, list2, list3, list4, list5, list6, list7):
    insurance_records = {}
    i = 0
    id = 1
    length = len(age)
    while i < length:
        id_string = str(str(id).zfill(4))
        insurance_records[id_string] = {"Age": list1[i], "Sex": list2[i], "BMI": list3[i], "Children": list4[i], "Smoker": list5[i], "Region": list6[i], "Charges": list7[i]}
        id += 1
        i += 1
    return insurance_records

insurance_records = populate_dict(age_int, sex, bmi_float, children_int, smoker, region, charges_float)


We are interested in finding out more about the dataset we're working with. First, let's check if the data is skewed in any way. To do this, we want to calculate the male vs female distribution, the age distribution, and region distribution. 

1. Calculate the average age and standard deviation of the dataset. 
2. Find out whether the majority of entries is 'male' or 'female'.
3. Find out how even the 'region' distribution is within the dataset. 

In [7]:
import statistics

# Functions to calculate age average and std deviation
def get_average(list):
    age_sum = 0
    for age in list:
        age_sum += age
    avg_age = round(age_sum / len(list), 2)
    median_age = round(statistics.median(list), 2)
    return "The average age within the data set is " + str(avg_age) + " years. The median age is " + str(median_age) + "."

age_average = get_average(age_int)

def get_stdev(list):
    stdev = round(statistics.stdev(list), 2)
    return "The standard deviation within the dataset is " + str(stdev) + "."

age_stdev = get_stdev(age_int)

# max/min age and age range 
min_age = min(age_int)
max_age = max(age_int)
age_range = max_age - min_age

print("The youngest person in the dataset is " + str(min_age) + " years old. The oldest person in the dataset is " + str(max_age) + " years old. The age range is " + str(age_range) + ".")
print(age_average)
print(age_stdev)

The youngest person in the dataset is 18 years old. The oldest person in the dataset is 64 years old. The age range is 46.
The average age within the data set is 39.21 years. The median age is 39.0.
The standard deviation within the dataset is 14.05.


Based on the age analysis, the dataset appears to be balanced. Let's take a look at male vs female entries. 

In [8]:
# Function to count male and female individuals
def counter(list, string):
    count = 0
    for item in list:
        if item == string:
            count += 1
    return count

num_women = counter(sex, "female")
num_men = counter(sex, "male")
percent_women = round(num_women / len(sex) * 100, 2)
percent_men = round(num_men / len(sex) * 100, 2)

print("The dataset contains " + str(percent_women) + " percent female and " + str(percent_men) + " percent male individuals.")

The dataset contains 49.48 percent female and 50.52 percent male individuals.


The dataset is ever so slightly skewed male, but still balanced overall. 

In [9]:
# Function to determine region balance
def find_unique_values(list):
    unique_values = []
    for item in list:
        if item not in unique_values:
            unique_values.append(item)
        else:
            continue
    return unique_values

unique_regions = find_unique_values(region)
print(unique_regions) 

['southwest', 'southeast', 'northwest', 'northeast']


In [10]:
# Function to count region distribution
def region_distribution(list):
    southwest = 0
    southeast = 0
    northwest = 0
    northeast = 0
    for item in list:
        if item == "southwest":
            southwest += 1
        elif item == "southeast":
            southeast += 1
        elif item == "northwest":
            northwest += 1
        elif item == "northeast":
            northeast += 1
    return "Southwest: " + str(southwest) + ", Southeast: " + str(southeast) + ", Northwest: " + str(northwest) + ", Northeast: " + str(northeast)

region_distribution(region)


'Southwest: 325, Southeast: 364, Northwest: 325, Northeast: 324'

There are 40 more entries from the region "Southeast" than from the other regions. Otherwise, the region distribution is even. 

Next, let's find out what the average charges are for the entire dataset. We also want to look at the minimum and maximum charges, the median, and the standard deviation, to find out if the data is skewed in any direction. 

In [11]:
def calculate_average(list):
    average_charges = round(sum(list) / len(list), 2)
    return average_charges

average_charges = calculate_average(charges_float)
median_charges = statistics.median(charges_float)
stddev_charges = get_stdev(charges_float)
print("The average charges are " + str(average_charges) + " dollars.")
print("The median charges are " + str(median_charges) + " dollars.")
print(stddev_charges)

The average charges are 13270.42 dollars.
The median charges are 9382.033 dollars.
The standard deviation within the dataset is 12110.01.


In [12]:
min_charges = round(min(charges_float), 2)
max_charges = round(max(charges_float), 2)
min_diff_avg = round(average_charges - min_charges, 2)
max_diff_avg = round(max_charges - average_charges, 2)

print("The lowest charges in the data set are " + str(min_charges) + " dollars. The highest charges are " + str(max_charges) + " dollars. The lowest charge is " + str(min_diff_avg) + " dollars lower than the average, while the highest charges are " + str(max_diff_avg) + " dollars higher than the average.")

The lowest charges in the data set are 1121.87 dollars. The highest charges are 63770.43 dollars. The lowest charge is 12148.55 dollars lower than the average, while the highest charges are 50500.01 dollars higher than the average.


Based on the distribution of charges, we can determine that the dataset contains some extreme outliers. This explains the high standard deviation, as well as the average that is positively skewed compared to the median. Let's find out where these extremes may be coming from by analyzing some other aspects. 

First, let's find out the average charges for smokers versus non-smokers. 

In [13]:
def group_by_smoker_status(smoker_status, charges):
    sum_smokers = 0.0
    num_smokers = 0
    sum_non_smokers = 0.0
    num_non_smokers = 0
    i = 0
    length = len(charges)
    while i < length:
        if smoker_status[i] == "yes":
            sum_smokers += float(charges[i])
            num_smokers += 1
            i += 1
        else:
            sum_non_smokers += float(charges[i])
            num_non_smokers += 1
            i += 1
    avg_charges_smokers = round(sum_smokers / num_smokers, 2)
    avg_charges_non_smokers = round(sum_non_smokers / num_non_smokers, 2)
    return "The average charge for smokers is " + str(avg_charges_smokers) + " dollars, and the average charge for non-smokers is " + str(avg_charges_non_smokers) + " dollars."

group_by_smoker_status(smoker, charges)

'The average charge for smokers is 32050.23 dollars, and the average charge for non-smokers is 8434.27 dollars.'

In [14]:
avg_charges_smokers = 32050.23
avg_charges_non_smokers = 8434.27

We can see that the average charges for smokers are significantly higher than for non-smokers. If we remove the smokers from the dataset, will the average charges come closer to the median, and can we reduce the standard deviation? 

In [15]:
# Function to remove smokers from dataset
def remove_smokers(smoker_status, charges):
    only_non_smokers = []
    i = 0
    length = len(charges)
    while i < length:
        if smoker_status[i] == "no":
            only_non_smokers.append(round(float(charges[i]), 2))
            i += 1
        else:
            i += 1
    return only_non_smokers

charges_non_smokers = remove_smokers(smoker, charges)

average_charges_non_smokers = calculate_average(charges_non_smokers)
median_non_smokers = round(statistics.median(charges_non_smokers), 2)
stddev_non_smokers = round(statistics.stdev(charges_non_smokers), 2)
print("The average charges for non-smokers is " + str(average_charges_non_smokers) + " dollars.")
print("The average charges for non-smokers are very close to the median charges of " + str(median_charges) + " dollars. The median within only non-smokers is " + str(median_non_smokers) + " dollars. The standard deviation among non-smokers is " + str(stddev_non_smokers) + ".")


The average charges for non-smokers is 8434.27 dollars.
The average charges for non-smokers are very close to the median charges of 9382.033 dollars. The median within only non-smokers is 7345.4 dollars. The standard deviation among non-smokers is 5993.78.


We can see that the data in the data set is strongly skewed by smokers, as their charges are much higher on average than the charges for non-smokers. However, there are other factors such as BMI and whether or not an individual has children, which also influence the charges. We will look at those factors later on. 

Next, we'll want to take a look at the average charges for male and female individuals, and how they compare to the median and mean charges. 

In [16]:
# Functions to separate male and female charges
def separate_by_sex(sex, charges):
    male_charges = []
    female_charges = []
    i = 0
    length = len(charges)
    while i < length:
        if sex[i] == "female":
            female_charges.append(float(charges[i]))
        else:
            male_charges.append(float(charges[i]))
        i += 1
    return male_charges, female_charges

male_charges_list, female_charges_list = separate_by_sex(sex, charges_float)

average_charges_male = round(sum(male_charges_list) / len(male_charges_list), 2)
average_charges_female = round(sum(female_charges_list) / len(female_charges_list), 2)
average_charges_male_diff = round(average_charges_male - average_charges, 2)
average_charges_female_diff = round(average_charges - average_charges_female, 2)
print("The average charges for male individuals is " + str(average_charges_male) + " dollars. This is " + str(average_charges_male_diff) + " higher than the average charges of " + str(average_charges) + " dollars.")
print("The average charges for female individuals is " + str(average_charges_female) + " dollars. This is " + str(average_charges_female_diff) + " lower than the average charges of " + str(average_charges) + " dollars.")

The average charges for male individuals is 13956.75 dollars. This is 686.33 higher than the average charges of 13270.42 dollars.
The average charges for female individuals is 12569.58 dollars. This is 700.84 lower than the average charges of 13270.42 dollars.


The results show that charges for males skew higher than average, while charges for females skew lower than average. However, there isn't as much of a skew as compared to the smokers vs non-smokers. 

Next, we'll want to look at the impact children have on the average charges. 

In [17]:
# Function to separate childless and individuals with children
def separate_by_children(children, charges):
    with_child = []
    without_child = []
    i = 0
    length = len(charges)
    while i < length:
        if children[i] == "0":
            without_child.append(float(charges[i]))
        else:
            with_child.append(float(charges[i]))
        i += 1
    return with_child, without_child

with_children_list, without_children_list = separate_by_children(children, charges_float)

average_charges_with_children = round(sum(with_children_list) / len(with_children_list), 2)
average_charges_without_children = round(sum(without_children_list) / len(without_children_list), 2)
print("The average charges for people with children is " + str(average_charges_with_children) + " dollars.")
print("The average charges for people without children is " + str(average_charges_without_children) + " dollars.")


The average charges for people with children is 13949.94 dollars.
The average charges for people without children is 12365.98 dollars.


The impact of having children on an individuals charges are not as pronounced as expected. In fact, they are similar to the difference between male and female individuals within the dataset. 

Next, we'll want to split the dataset into age groups and calculate the respective averages. 

In [18]:
def separate_by_age(age, charges):
    under_thirty = []
    thirty_to_forty = []
    forty_to_fifty = []
    fifty_and_over = []
    i = 0
    length = len(charges)
    while i < length:
        if age[i] < 30:
            under_thirty.append(float(charges[i]))
        elif age[i] < 40:
            thirty_to_forty.append(float(charges[i]))
        elif age[i] < 50:
            forty_to_fifty.append(float(charges[i]))
        else:
            fifty_and_over.append(float(charges[i]))
        i += 1
    return under_thirty, thirty_to_forty, forty_to_fifty, fifty_and_over

under_thirty, thirty_to_forty, forty_to_fifty, fifty_and_over = separate_by_age(age_int, charges_float)

average_under_thirty = round(sum(under_thirty) / len(under_thirty), 2)
average_thirty_to_forty = round(sum(thirty_to_forty) / len(thirty_to_forty), 2)
average_forty_to_fifty = round(sum(forty_to_fifty) / len(forty_to_fifty), 2)
average_fifty_and_over = round(sum(fifty_and_over) / len(fifty_and_over), 2)
print("The average charges for individuals under thirty are " + str(average_under_thirty) + " dollars.")
print("The average charges for individuals in their thirties are " + str(average_thirty_to_forty) + " dollars.")
print("The average charges for individuals in their forties are " + str(average_forty_to_fifty) + " dollars.")
print("The average charges for individuals over fifty are " + str(average_fifty_and_over) + " dollars.")

The average charges for individuals under thirty are 9182.49 dollars.
The average charges for individuals in their thirties are 11738.78 dollars.
The average charges for individuals in their forties are 14399.2 dollars.
The average charges for individuals over fifty are 17902.55 dollars.


We can see that the average charges are higher in the older age brackets. So far, it appears that the following tend to push the average charges higher: sex=male, age>30, smoker=yes, and children>0. In order to verify these indications, let's separate out all entries with high charges and analyze their commonalities. 

In order to define "high charges", let's calculate the quartiles and pick the highest quartile for our analysis.    

In [26]:
import numpy

charges_quartiles = numpy.quantile(charges_float, [0, 0.25, 0.5, 0.75, 1])
print(charges_quartiles)


[ 1121.8739    4740.28715   9382.033    16639.912515 63770.42801 ]


The fourth quartile of the data starts at charges of 16639.91 and higher, so let's take all charges that are higher than that from the charges_float list and go from there. 

In [43]:
def create_fourth_quartile_records(charges, age, sex, bmi, children, smoker, region):
    charges_fq = []
    age_fq = []
    sex_fq = []
    bmi_fq = []
    children_fq = []
    smoker_fq = []
    region_fq = []
    i = 0
    length = len(charges)
    while i < length: 
        if charges[i] >= 16639:
            charges_fq.append(charges[i])
            age_fq.append(age[i])
            sex_fq.append(sex[i])
            bmi_fq.append(bmi[i])
            children_fq.append(children[i])
            smoker_fq.append(smoker[i])
            region_fq.append(region[i])
        i += 1
    return charges_fq, age_fq, sex_fq, bmi_fq, children_fq, smoker_fq, region_fq

charges_fq, age_fq, sex_fq, bmi_fq, children_fq, smoker_fq, region_fq = create_fourth_quartile_records(charges_float, age_int, sex, bmi_float, children_int, smoker, region)





We've now created lists for each attribute for only the individuals whose charges are in the highest quartile of the dataset. Let's compare averages within this subset of high payers to the overall averages. 

In [58]:
# Compare age
print(age_average)

age_avg_fq = round(sum(age_fq) / len(age_fq), 2)
print("The average age within the fourth quartile is " + str(age_avg_fq) + " years.\n")

# Compare sex
print("The entire dataset contains " + str(percent_women) + " percent female and " + str(percent_men) + " percent male individuals.")

num_women_fq = counter(sex_fq, "female")
num_men_fq = counter(sex_fq, "male")
percent_women_fq = round(num_women_fq / len(sex_fq) * 100, 2)
percent_men_fq = round(num_men_fq / len(sex_fq) * 100, 2)
print("The fourth quartile contains " + str(percent_women_fq) + " percent female and " + str(percent_men_fq) + " percent male individuals.\n")

# Compare bmi
average_bmi = round(sum(bmi_float) / len(bmi_float), 2)
print("The average BMI within the entire dataset is " + str(average_bmi) + ".")
average_bmi_fq = round(sum(bmi_fq) / len(bmi_fq), 2)
print("The average BMI in the fourth quartile is " + str(average_bmi_fq) + ".\n")

# Compare childless vs with children
num_of_childless = children_int.count(0)
num_with_children = len(children_int) - num_of_childless
percent_with_children = round(num_with_children / len(children_int) * 100, 2)
print(str(percent_with_children) + " percent of individuals in the entire dataset have children.")

num_of_childless_fq = children_fq.count(0)
num_with_children_fq = len(children_fq) - num_of_childless_fq
percent_with_children_fq = round(num_with_children_fq / len(children_fq) * 100, 2)
print(str(percent_with_children_fq) + " percent of individuals in the fourth quartile have children.\n")

# Compare smoker status
percent_of_smokers = round(smoker.count("yes") / len(smoker) * 100, 2)
print(str(percent_of_smokers) + " percent of individuals in the entire dataset are smokers.")

percent_of_smokers_fq = round(smoker_fq.count("yes") / len(smoker_fq) * 100, 2)
print(str(percent_of_smokers_fq) + " percent of individuals in the fourth quartile are smokers.")

The average age within the data set is 39.21 years. The median age is 39.0.
The average age within the fourth quartile is 40.19 years.

The entire dataset contains 49.48 percent female and 50.52 percent male individuals.
The fourth quartile contains 43.88 percent female and 56.12 percent male individuals.

The average BMI within the entire dataset is 30.66.
The average BMI in the fourth quartile is 31.31.

57.1 percent of individuals in the entire dataset have children.
61.49 percent of individuals in the fourth quartile have children.

20.48 percent of individuals in the entire dataset are smokers.
76.12 percent of individuals in the fourth quartile are smokers.


Seeing as most averages are consistent with the entire dataset, but the rate of smokers among the fourth quartile is 55 % higher than in the entire dataset, it becomes clear that smoking appears to contribute the most to high insurance charges. 

Let's now look at the opposite end, the first quartile of charges, i.e., the individuals with the lowest insurance charges, and their commonalities. 

In [69]:
def create_first_quartile_records(charges, age, sex, bmi, children, smoker, region):
    charges_frstq = []
    age_frstq = []
    sex_frstq = []
    bmi_frstq = []
    children_frstq = []
    smoker_frstq = []
    region_frstq = []
    i = 0
    length = len(charges)
    while i < length: 
        if charges[i] <= 4741:
            charges_frstq.append(charges[i])
            age_frstq.append(age[i])
            sex_frstq.append(sex[i])
            bmi_frstq.append(bmi[i])
            children_frstq.append(children[i])
            smoker_frstq.append(smoker[i])
            region_frstq.append(region[i])
        i += 1
    return charges_frstq, age_frstq, sex_frstq, bmi_frstq, children_frstq, smoker_frstq, region_frstq

charges_frstq, age_frstq, sex_frstq, bmi_frstq, children_frstq, smoker_frstq, region_frstq = create_first_quartile_records(charges_float, age_int, sex, bmi_float, children_int, smoker, region)

In [75]:
# Compare age
print(age_average)

age_avg_frstq = round(sum(age_frstq) / len(age_frstq), 2)
print("The average age within the first quartile is " + str(age_avg_frstq) + " years.\n")

# Compare sex
print("The entire dataset contains " + str(percent_women) + " percent female and " + str(percent_men) + " percent male individuals.")

num_women_frstq = counter(sex_frstq, "female")
num_men_frstq = counter(sex_frstq, "male")
percent_women_frstq = round(num_women_frstq / len(sex_frstq) * 100, 2)
percent_men_frstq = round(num_men_frstq / len(sex_frstq) * 100, 2)
print("The first quartile contains " + str(percent_women_frstq) + " percent female and " + str(percent_men_frstq) + " percent male individuals.\n")

# Compare bmi
print("The average BMI within the entire dataset is " + str(average_bmi) + ".")
average_bmi_frstq = round(sum(bmi_frstq) / len(bmi_frstq), 2)
print("The average BMI in the first quartile is " + str(average_bmi_frstq) + ".\n")

# Compare childless vs with children
print(str(percent_with_children) + " percent of individuals in the entire dataset have children.")

num_of_childless_frstq = children_frstq.count(0)
num_with_children_frstq = len(children_frstq) - num_of_childless_frstq
percent_with_children_frstq = round(num_with_children_frstq / len(children_frstq) * 100, 2)
print(str(percent_with_children_frstq) + " percent of individuals in the first quartile have children.\n")

# Compare smoker status
print(str(percent_of_smokers) + " percent of individuals in the entire dataset are smokers.")

percent_of_smokers_frstq = round(smoker_frstq.count("yes") / len(smoker_frstq) * 100, 2)
print(str(percent_of_smokers_frstq) + " percent of individuals in the first quartile are smokers.")

The average age within the data set is 39.21 years. The median age is 39.0.
The average age within the first quartile is 23.81 years.

The entire dataset contains 49.48 percent female and 50.52 percent male individuals.
The first quartile contains 47.76 percent female and 52.24 percent male individuals.

The average BMI within the entire dataset is 30.66.
The average BMI in the first quartile is 29.91.

57.1 percent of individuals in the entire dataset have children.
39.1 percent of individuals in the first quartile have children.

20.48 percent of individuals in the entire dataset are smokers.
0.0 percent of individuals in the first quartile are smokers.


Perhaps unsurprisingly, the first quartile contains absolutely no smokers, and has an average age of 23.81, which is about 16 years less than the average in the entire dataset. 

Lastly, let's look at whether or not the region individuals are located in has any correlation with their average charges. We don't know anything about living conditions or external health factors for each region, but if there are large discrepancies between the regions, it might be useful to find out if factors such as those exist. 

We will:

1. Group the entries by region. To do this, we will define a create_subset function that takes two lists and a region marker as arguments, and we'll create separate lists for each attribute in each region. 
2. Then, we will compare the averages between these regions and draw conclusions. 

In [77]:
def create_subset(list, region_list, region_marker):
    output_list = []
    i = 0
    length = len(list)
    while i < length:
        if region_list[i] == region_marker:
            output_list.append(list[i])
        i += 1
    return output_list

In [81]:
ne_age = create_subset(age_int, region, "northeast")
ne_sex = create_subset(sex, region, "northeast")
ne_bmi = create_subset(bmi_float, region, "northeast")
ne_children = create_subset(children_int, region, "northeast")
ne_smoker = create_subset(smoker, region, "northeast")
ne_charges = create_subset(charges_float, region, "northeast")

se_age = create_subset(age_int, region, "southeast")
se_sex = create_subset(sex, region, "southeast")
se_bmi = create_subset(bmi_float, region, "southeast")
se_children = create_subset(children_int, region, "southeast")
se_smoker = create_subset(smoker, region, "southeast")
se_charges = create_subset(charges_float, region, "southeast")

sw_age = create_subset(age_int, region, "southwest")
sw_sex = create_subset(sex, region, "southwest")
sw_bmi = create_subset(bmi_float, region, "southwest")
sw_children = create_subset(children_int, region, "southwest")
sw_smoker = create_subset(smoker, region, "southwest")
sw_charges = create_subset(charges_float, region, "southwest")

nw_age = create_subset(age_int, region, "northwest")
nw_sex = create_subset(sex, region, "southwest")
nw_bmi = create_subset(bmi_float, region, "southwest")
nw_children = create_subset(children_int, region, "southwest")
nw_smoker = create_subset(smoker, region, "southwest")
nw_charges = create_subset(charges_float, region, "southwest")


In [91]:
# Age average by region
# Region: northeast
ne_age_avg = round(sum(ne_age) / len(ne_age), 2)
print("The age average in the northeast is " + str(ne_age_avg) + ".")

# Region: southeast
se_age_avg = round(sum(se_age) / len(se_age), 2)
print("The age average in the southeast is " + str(se_age_avg) + ".")

# Region: southwest
sw_age_avg = round(sum(sw_age) / len(sw_age), 2)
print("The age average in the southwest is " + str(sw_age_avg) + ".")

# Region: northwest
nw_age_avg = round(sum(nw_age) / len(nw_age), 2)
print("The age average in the northwest is " + str(nw_age_avg) + ".\n\n")

# Sex distribution by region
# ne
ne_percent_of_women = round(ne_sex.count("female") / len(ne_sex) * 100, 2)
print(str(ne_percent_of_women) + " percent of individuals in the northeast are female.")
ne_percent_of_men = round(ne_sex.count("male") / len(ne_sex) * 100, 2)
print(str(ne_percent_of_men) + " percent of individuals in the northeast are male.\n")

# se
se_percent_of_women = round(se_sex.count("female") / len(se_sex) * 100, 2)
print(str(se_percent_of_women) + " percent of individuals in the southeast are female.")
se_percent_of_men = round(se_sex.count("male") / len(se_sex) * 100, 2)
print(str(se_percent_of_men) + " percent of individuals in the southeast are male.\n")

# sw
sw_percent_of_women = round(sw_sex.count("female") / len(sw_sex) * 100, 2)
print(str(sw_percent_of_women) + " percent of individuals in the southwest are female.")
sw_percent_of_men = round(sw_sex.count("male") / len(sw_sex) * 100, 2)
print(str(sw_percent_of_men) + " percent of individuals in the southwest are male.\n")

# nw
nw_percent_of_women = round(nw_sex.count("female") / len(nw_sex) * 100, 2)
print(str(nw_percent_of_women) + " percent of individuals in the northwest are female.")
nw_percent_of_men = round(nw_sex.count("male") / len(nw_sex) * 100, 2)
print(str(nw_percent_of_men) + " percent of individuals in the northwest are male.\n\n")

# Average charges per region
# ne
ne_avg_charges = round(sum(ne_charges) / len(ne_charges), 2)
print("The average charges in the northeast are " + str(ne_avg_charges) + " dollars.")

# se
se_avg_charges = round(sum(se_charges) / len(se_charges), 2)
print("The average charges in the southeast are " + str(se_avg_charges) + " dollars.")

# sw
sw_avg_charges = round(sum(sw_charges) / len(sw_charges), 2)
print("The average charges in the southwest are " + str(sw_avg_charges) + " dollars.")

# nw
nw_avg_charges = round(sum(nw_charges) / len(nw_charges), 2)
print("The average charges in the northwest are " + str(nw_avg_charges) + " dollars.")

The age average in the northeast is 39.27.
The age average in the southeast is 38.94.
The age average in the southwest is 39.46.
The age average in the northwest is 39.2.


49.69 percent of individuals in the northeast are female.
50.31 percent of individuals in the northeast are male.

48.08 percent of individuals in the southeast are female.
51.92 percent of individuals in the southeast are male.

49.85 percent of individuals in the southwest are female.
50.15 percent of individuals in the southwest are male.

49.85 percent of individuals in the northwest are female.
50.15 percent of individuals in the northwest are male.


The average charges in the northeast are 13406.38 dollars.
The average charges in the southeast are 14735.41 dollars.
The average charges in the southwest are 12346.94 dollars.
The average charges in the northwest are 12346.94 dollars.
